### PR TITLE
test: 5449 shard tests

### PR DESCRIPTION
**Pull request by tihuan**
## Reason for Change

- #5449 

WARNING ⚠️ :
~We had to upgrade to Playwright 1.37.0 to use the merge report function, however, `.only()` is temporarily broken at the moment. See [issue](https://github.com/microsoft/playwright/issues/26492)~ UPDATE: Upgraded to 1.37.1, which fixes the `.only()` issue!

## Changes

1. Upgrade Playwright to the latest (1.37.0)
2. Update e2e test to run with 30 workers
3. NOT running firefox and edge yet, because that's too many workers for GHA to handle. We will revisit testing FF and Edge down the road
4. Add GHA steps to upload and merge sharded Playwright reports

## Testing steps

1. e2e tests should now run with 30 works and roughly ~10 mins
2. You should still see the test report in the GHA summary artifacts section

### CODE REVIEW

File: .github/workflows/push-tests.yml [hunk 1]
```diff
@@ -119,6 +171,71 @@ jobs:
...
+        uses: actions/download-artifact@v3
+        with:
+          name: all-blob-reports
+          path: all-blob-reports
+
+      - uses: actions/checkout@v3
+      - name: Merge Playwright sharded reports
+        run: |
+          npx playwright --version
+          npx playwright merge-reports --reporter html ./all-blob-reports
```
Comment (tihuan) [inline .github/workflows/push-tests.yml:219]: command not found for some reason

Comment (tihuan) [reply .github/workflows/push-tests.yml:219]: Asking Playwright in Discord here: https://discord.com/channels/807756831384403968/1141146168299094107

**Comment by tihuan** [comment]
@seve  Not in a rush! We can park this PR until 1.38 is published 👍

**Comment by tihuan** [comment]
Parking this PR until Playwright 1.38 is out and hopefully `.only()` is fixed! https://github.com/microsoft/playwright/issues/26492

**Comment by tihuan** [comment]
fixing npm commands a little

**Review by atarashansky — APPROVED** [review]

**Pull request merged**